### PR TITLE
Fix card examples becoming empty when re-opening the app

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
Prevent from components being attached multiple times to `AdyenComponentView`.

Bonus fix: handle back press correctly in BACS

## Checklist 
- [x] Changes are tested manually

COAND-711

### CODE REVIEW

**Review by jreij — APPROVED** [review]

**Comment by etman55** [comment]
I think we should come up with another solution as we don't want for the flow to be observed when the app is in the background as `CREATYED` gets triggered just two time after `onCreate` and before `onStop` as mentioned in the [docs](https://developer.android.com/reference/android/arch/lifecycle/Lifecycle.State#created)

File: ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt [hunk 1]
```diff
@@ -59,7 +60,8 @@ class AdyenComponentView @JvmOverloads constructor(
     private var isInteractionBlocked = false
 
     private var componentView: ComponentView? = null
-    private var componentViewType: ComponentViewType? = null
+
+    private var attachedComponent = WeakReference<Component?>(null)
```
Comment (etman55) [inline ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt:64]: why did you remove `componentViewType`?

Comment (OscarSpruit) [reply ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt:64]: It wasn't used

**Review by jreij — APPROVED** [review]
